sources/en/r/ranobes.py

```diff
@@ -1,109 +1,125 @@
 # -*- coding: utf-8 -*-
 import logging
+import re
```

Comment:
Also, as an educational exercise, it might be interesting to try using [re2](https://github.com/google/re2), instead.

It's google's implementation of a regex [NFA engine](https://swtch.com/~rsc/regexp/regexp1.html), which is an [ultra-performant](https://swtch.com/~rsc/regexp/regexp1.html) [superset](https://github.com/google/re2/wiki/Syntax) of modern regex implementations, that means that you can use any kind off regex you want, and if it happens to break NFA compatibility, it'll just fall back to standard regex.

It doesn't make any noticeable difference from a user perspective in this specific case, since an url can only be so long. But in other projects of yours this could really come in handy when the data handled start getting into the dozens or hundreds of characters. Or simply because the regex is really complicated.

In practice, this is just a matter of:
- Importing the re2 wrapper as re.
```Python
import re2 as re
```
- Raising an import error in case the Python version is incompatible. (lower than version 2.6)
```Python
try:
    import re2 as re
except ImportError:
    raise ImportError('Python 2.6 or higher is needed for some speed optimizations.')
```
- And import re if re2 is unavailable
```Python
try:
    import re2 as re
except ImportError:
    raise ImportError('Python 2.6 or higher is needed for some speed optimizations.')
else:
    import re
```
Also, apparently, you should have this line just after importing re2:
```Python
re.set_fallback_notification(re.FALLBACK_WARNING)
```
Why? I have no idea, I'm not _that_ familiar with Python.

But, I hope this was useful to you and you learned something today.